crates/services/executor/src/ports.rs

```diff
@@ -55,8 +55,13 @@ impl MaybeCheckedTransaction {
 }
 
 pub trait TransactionsSource {
-    /// Returns the next batch of transactions to satisfy the `gas_limit`.
-    fn next(&self, gas_limit: u64) -> Vec<MaybeCheckedTransaction>;
+    /// Returns the next batch of transactions to satisfy the `gas_limit` and the
+    /// without selecting more than `max_transactions` transactions. Because
+    /// `max_transactions` has type `u16`, no more than `u16::MAX` transactions
+    /// will be included in a block, which enforce the property that all transaction
+    /// outputs will be indexable in the future.
+    fn next(&self, gas_limit: u64, max_transactions: u16)
+        -> Vec<MaybeCheckedTransaction>;
```

Comment:
IIUC Here we also want to take the number of slots to reserve for FTI transactions?